plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -87,24 +89,63 @@ export class DatabaseNotificationsStore implements NotificationsStore {
   };
 
   private mapToNotifications = (rows: any[]): Notification[] => {
-    return rows.map(row => ({
-      id: row.id,
-      user: row.user,
-      created: new Date(row.created),
-      saved: row.saved,
-      read: row.read,
-      updated: row.updated,
-      origin: row.origin,
-      payload: {
-        title: row.title,
-        description: row.description,
-        link: row.link,
-        topic: row.topic,
-        severity: row.severity,
-        scope: row.scope,
-        icon: row.icon,
-      },
-    }));
+    let order = 0;
+    const res = [
+      ...rows
```

Comment:
This array-spread-values-reduce nested thingamabob is kinda hard to read. I'd suggest to split it up like

```ts
// Roll up duplicate lines due to a JOIN with metadata
const idToNotification = new Map<string, { order: number, notification: Notification}>();
for (const row of rows) {
 ...
}

return [...idToNotification.values()].sort(...);
```

or, if the metadata became just a json blob as mentioned in a different comment, maybe the rollup isn't needed

Reply:
Sure, will split it, we will not come back to a blob because of this https://github.com/backstage/backstage/pull/24562/files/09cbfdea105498b1ee546aada16c366402a00e3a#r1601491520 😄 (it was my initial implementation with a block thingy)